src/yaml.cpp

```diff
@@ -105,31 +103,66 @@ estd::result<YAML::Node, estd::error> readYamlFile(std::string const & path) {
 	return YAML::Load(buffer.str());
 }
 
-YamlResult<void> mergeYamlNodes(YAML::Node & map_a, YAML::Node map_b) {
-	// Check if the arguments are maps.
-	if (!map_a.IsMap() && !map_a.IsNull()) {
-		return YamlError{"tried to merge into a YAML node that is not a map"};
-	}
-	if (!map_b.IsMap() && !map_b.IsNull()) {
-		return YamlError{"tried to merge from a YAML node that is not a map"};
-	}
-
+// Merge Yaml Nodes (Type: Map).
+YamlResult<void> mergeYamlMaps(YAML::Node & map_a, YAML::Node map_b) {
 	for (YAML::const_iterator iterator = map_b.begin(); iterator != map_b.end(); iterator++) {
 		std::string key = iterator->first.as<std::string>();
 		YAML::Node value = iterator->second;
-		if (value.IsMap() && map_a[key]) {
-			auto merged = mergeYamlNodes(map_a[key], value);
+
+		if (value.IsMap() && map_a[key] && map_a[key].IsMap()) {
+			auto merged = mergeYamlMaps(map_a[key], value);
 			if (!merged) {
 				merged.error().appendTrace({key, "", YAML::NodeType::Map});
 				return merged;
 			}
 			continue;
 		}
+		else if (value.Tag() == "!ordered_dict" && map_a[key]) {
+			auto merged = mergeYamlOrderedDict(map_a[key], value);
+			if(!merged) {
+				merged.error().appendTrace({key, "", YAML::NodeType::Map});
+				return merged;
+			}
+			continue;
+		}
 		map_a[key] = value;
```

Comment:
Not new, but if we use an `else` here we can remove all the `continue` statements before. Not super important though, either way is fine.

```suggestion
		} else {
			map_a[key] = value;
		}
```